### DISCUSSION TITLE
Windows Terminal v1.12.10334.0

### DISCUSSION THREAD

**Discussion by DHowett** [microsoft/terminal#12346] [Announcements]
Happy New Year!

This release brings many of the preview changes in Windows Terminal 1.12 to the Stable channel. Notably:

* On Windows 11, Terminal now supports non-acrylic transparency
* Terminal will now automatically create profiles for Visual Studio Developer shells
* Terminal will now try to guess which profile to open when it is launched as the ‘default terminal’ for an application
* You can now update the endpoint of an existing selection using the keyboard

Please see the following release notes for additional details:

* [Windows Terminal Preview v1.12.2922.0](https://github.com/microsoft/terminal/releases/tag/v1.12.2922.0)
* [Windows Terminal Preview v1.12.3472.0](https://github.com/microsoft/terminal/releases/tag/v1.12.3472.0)

A [preinstallation](https://docs.microsoft.com/en-us/windows/msix/desktop/deploy-preinstalled-apps) kit is available for system integrators and OEMs interested in prepackaging Windows Terminal with a Windows image. More information is available in the [DISM documentation on preinstallation](https://docs.microsoft.com/windows-hardware/manufacture/desktop/preinstall-apps-using-dism). Users who do not intend to preinstall Windows Terminal should continue using the _msixbundle_ distribution.

This release was made available to insiders in the Dev and Beta channels on February 3, and will be generally available to everyone shortly afterwards. You can always install the update from this release page or using `winget`.

We've also backported the following features, changes and bug fixes from [Windows Terminal Preview v1.13.10336.0](https://github.com/microsoft/terminal/releases/tag/v1.13.10336.0):

## Announcements

* _Windows Terminal_ is now named _Terminal_! It will, of course, still be called `wt` under the hood. (#12264)
   * _Right now, you will not be able to find_ Windows Terminal _ (that is, with the "Windows" part) in the start menu. We are working with the search team to add a keyword match! Follow #12344 for updates._
   * Sorry about your muscle memory. :(
* **1.12 is the last version of Terminal that will support Windows 19H1 or 19H2** (#12129)
   * Folks who are on the Preview channel _on these versions of Windows_ will get a one-time upgrade to the last servicing release of 1.12.
   * This version of windows is going out of support soon, so you may want to consider upgrading.

## Changes

* We're reworking how Terminal handles environment variables, starting with reverting the changes we made in
  Preview that broke PowerShell x86 (#12140)
* The "adjust lightness of indistinguishable text" feature, present in 1.12 preview, has been disabled pending
  some bug fixes (#12160)
* The "Open in Terminal" context menu is now localized in, like, 88 languages (#12090)
* The PowerShell profile generator now generates fully quoted paths (#12086) (thanks @ianjoneill!)
* WSL instances now default to starting in `~` (or throwing an `Invalid Parameter` error) (#12315)

## Bug Fixes

* The cursor should now start/stop blinking more reliably when Terminal is started without window focus (#12094)
* Copying the "Azure Cloud Shell" profile will no longer result in a normal boring profile that tries to run `Azure.exe` (lol) (#12147)
* The default "Command Prompt" and "Windows PowerShell" profiles have been updated to have fully qualified paths (again) (#12149)
* Cursor movement will now more reliably update the IME/emoji picker/input line/cursor highlight for Accessibility (#12210)
* HTML/RTF copy now respects the active background color instead of just the default one (#11991) (thanks @j4james!)
* The "Duplicate Profile" button will no longer be enabled when you don't actually have a profile selected (#12096) (thanks @DavideGiacometti!)
* `//wsl$` paths will no longer result in you being unceremoniously dumped in `/` (in WSL) (#12102)
* `~` now works as a starting directory for `wsl.exe` (#12050) (thanks @LuanVSO!)
* Application-controlled titles will be sanitized for control sequences _even if they are set with the `SetConsoleTitle` API_ (#12211) (thanks @j4james!)

### Accessibility

* The command palette will now announce if any suggestions were found to screen readers (#12266)
* ... so will the search box! (#12301)
* The tab row/titlebar have stopped acting as a keyboard input/focus trap -- you can use keyboard shortcuts up there now (#12260)
* We've done a pass over the controls in the settings UI and given them accessible names for screen reader users (#12299) (#12324)
* The "Default Terminal" section will now properly read out terminal names to Narrator (#12259)
* "Open JSON File" is now a better accessibility citizen as well (#12286)

## Documentation Changes

We'd like to thank @eltociear, @BlakeHeimann, @ghost1372, @Schweinepriester, and @ianjoneill for their contributions
to our documentation, roadmap, specs and schema.

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/terminal/releases/tag/v1.12.10334.0'>Windows Terminal v1.12.10334.0</a>.</em>

**Comment by shtrophic** [comment]
> On Windows 11, Terminal now supports non-acrylic transparency

https://www.youtube.com/watch?v=P3ALwKeSEYs

**Comment by dotnetCarpenter** [comment]
@DHowett Windows Terminal v1.12.10334.0 is still not pushed to Microsoft Store as of 8th of February 2022.

I got Windows Terminal Version: 1.11.3471.0 as the latest version available in Microsoft Store Version 22112.1401.2.0

**Reply by zadjii-msft** [reply]
This unfortunately happens [every](https://github.com/microsoft/terminal/issues/11622) [release](https://github.com/microsoft/terminal/issues/12403). The Store doesn't necessarily roll out the updates uniformly. If you install the `.msixbundle` off our [Releases page](https://github.com/microsoft/terminal/releases), then it'll stay linked to the Store version and still auto-update in the future, if you manually want to hop to the next version without waiting for the Store.

**Reply by DHowett** [reply]
In addition, we use a slow rollout strategy for the Stable channel because we have a couple _million_ users. We have shipped nearly-catastrophic bugs in the past, and this helps prevent us from doing that in the future.

For example: #12351 would have switched Terminal to English for multiple millions of people, all at once, and been the cause of a significant disaster.

**Reply by DHowett** [reply]
I mistakenly said that we shipped this to insiders on December 14! That should have said "February 3" 😅

**Reply by dotnetCarpenter** [reply]
Thanks to both of you, for the explanation. I will install the `.msixbundle` the next time I get impatience. Ironically, I opted for the Store release because I didn't want to miss a new version.

@DHowett whoa good anecdote 😃

**Reply by joshsleeper** [reply]
apologies for the bump and noise, but nearly a full month after shipping `1.12` to stable I'm still not seeing the update through Windows Store yet.
I know I can install the `.msixbundle` at my convenience, but I just wanted to ask if that's still expected?

**Reply by zadjii-msft** [reply]
Yep. We're tracking down a couple crashes that we're seeing from the latest build and making sure they get cleaned up before rolling it out to the broader population.

**Reply by clicktodev** [reply]
@zadjii-msft Can you link us the open issues about the crashes, I wanna make sure It's not something too severe before manually updating using the `.msixbundle`.

**Reply by zadjii-msft** [reply]
As much as is possible: #12624, #12580. So, these crashes are super trivial. 
* The first group of crashes only seems to happen when you've got the Terminal set up as the default terminal emulator, in which case the Terminal crashing would cause the OS to fall back to using conhost as the terminal. So, not so bad. 
* The second group entirely seems to happen when the Terminal is _already closing_. So from the end user perspective, nothing goes wrong - you click the 'x', and the Terminal closes. Only problem is, it still technically crashed. 

So IMO neither of these are all that bad, but they do need to get cleaned up.

**Reply by clicktodev** [reply]
Not sure if I should open a new issue for this or not but the ms store terminal has been stuck on version 1.11.3471.0 for a long time now.
Running windows 10, is w11 required to get newer versions of wt?

**Reply by DHowett** [reply]
We finally published 1.12.1073 to the store for all users. Thank you for your patience!

**Reply by clicktodev** [reply]
Just got it, thanks. opening wsl Ubuntu in wt was broken after the update (something about `The parameter is incorrect`) but I fixed it by resetting the app.
